pilot/pkg/model/network.go

```diff
@@ -108,40 +155,101 @@ func NewNetworkManager(env *Environment) *NetworkManager {
 		lcmVal = lcm(lcmVal, num)
 	}
 
-	return &NetworkManager{
-		lcm:                 uint32(lcmVal),
-		byNetwork:           byNetwork,
-		byNetworkAndCluster: byNetworkAndCluster,
+	mgr.lcm = uint32(lcmVal)
+	mgr.byNetwork = byNetwork
+	mgr.byNetworkAndCluster = byNetworkAndCluster
+
+	return gatewaySet
+}
+
+func (mgr *NetworkManager) resolveHostnameGateways(gatewaySet map[NetworkGateway]struct{}) {
+	// filter the list of gateways to resolve
+	hostnameGateways := map[string][]NetworkGateway{}
+	names := sets.NewSet()
+	for gw := range gatewaySet {
+		if gwIP := net.ParseIP(gw.Addr); gwIP != nil {
+			continue
+		}
+		delete(gatewaySet, gw)
+		if !features.HostnameGateways {
+			log.Warnf("Failed parsing gateway address %s from Service Registry. "+
+				"Hostnames are not supported for gateways",
+				gw.Addr)
+			continue
+		}
+		hostnameGateways[gw.Addr] = append(hostnameGateways[gw.Addr], gw)
+		names.Insert(gw.Addr)
+	}
+
+	// resolve each hostname
+	for host, addrs := range mgr.nameCache.Resolve(names) {
+		gwsForHost := hostnameGateways[host]
+		if len(addrs) == 0 {
+			log.Warnf("could not resolve hostname %q for %d gateways", host, len(gwsForHost))
```

Comment:
What is the implication of this?  data plane traffic through that gw won't function?

Reply:
or no traffic would be sent to that gw?

Reply:
Right, we won't have an endpoint for that gateway so we drop it. This likely means that the control plane can't hit whichever nameservers actually resolve the hostname.